extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPuller.java

```diff
@@ -273,6 +276,65 @@ public boolean apply(Throwable e)
   }
 
   /**
+   * Gets the latest version using the "parent" of uri as a prefix. The "parent" of uri is evaluated assuming '/' delimited paths
+   *
+   * @param uri     The URI of interest whose "parent" will be searched as a key prefix for the latest version
+   * @param pattern The pattern matcher to determine if a *key* is of interest. This will match against the entire key,
+   *                not just the equivalent "filename" like some other implementations. A null value matches everything
+   *
+   * @return A URI to the most recently modified object which matched the pattern.
+   */
+  @Override
+  public URI getLatestVersion(final URI uri, final Pattern pattern)
+  {
+    try {
+      return RetryUtils.retry(
+          new Callable<URI>()
+          {
+            @Override
+            public URI call() throws Exception
+            {
+              final AmazonS3URI s3URI = new AmazonS3URI(checkURI(uri));
+              long mostRecent = Long.MIN_VALUE;
+              URI latest = null;
+              String parentPath = s3URI.getKey().endsWith("/")
+                                ? s3URI.getKey()
+                                : Paths.get(s3URI.getKey()).getParent().toString();
+              if(!parentPath.endsWith("/")){
+                parentPath = parentPath + "/";
+              }
+              S3Object[] objects = s3Client.listObjects(s3URI.getBucket(), parentPath, "/");
+              if (objects == null) {
+                return null;
+              }
+              for (S3Object storageObject : objects) {
+                String keyString = storageObject.getKey().substring(parentPath.length());
+                if (keyString.startsWith("/")) {
+                  keyString = keyString.substring(1);
+                }
+                if (pattern != null && !pattern.matcher(keyString).matches()) {
+                  continue;
+                }
+                final long latestModified = storageObject.getLastModifiedDate().getTime();
+                if (latestModified >= mostRecent) {
+                  mostRecent = latestModified;
+                  // TOOD: Does this need to change for non-standard endpoints?
```

Comment:
I looked into it and no, it seems to be a property setting on jets3t rather than a property that needs accounted for here.